### PR TITLE
DOC: update 0.4.0 release notes with deprecations from gh-55.

### PR DESCRIPTION
**Pull request by rgommers**

### CODE REVIEW

**Comment by aaren** [comment]
Looks good to me.

File: doc/release/0.4.0-notes.rst [hunk 1]
```diff
@@ -18,6 +18,21 @@ New features
...
+- ``centrfrq``, renamed to ``central_frequency``
+- ``scal2frq``, renamed to ``scale2frequency``
+- ``orthfilt``, renamed to ``orthogonal_filter_bank``
+
+Integration of general signals (i.e. not wavelets) with ``integrate_wavelet``
+is deprecated.
+
+The ``MODES`` object and its attributes are deprecated.  The new name is
+``Modes``, and the attribute names are expanded: ``zero``, ``constant``,
+``symmetric``, ``periodic``, ``smooth`` and ``periodization``.
```
Comment (aaren) [inline doc/release/0.4.0-notes.rst]: Would be useful to include the old names as well:

```
- ``zpd``, renamed to ``zero``
- ``sp0``, renamed to ``constant``
- ``sp1``, renamed to ``smooth``
- ``sym``, renamed to ``symmetric``
- ``ppd``, renamed to ``periodic``
- ``per``, renamed to ``periodization``
```

Comment (rgommers) [inline doc/release/0.4.0-notes.rst]: good idea, done. note that ``constant`` was ``cpd``, not ``sp0``.

Comment (aaren) [inline doc/release/0.4.0-notes.rst]: good catch. `sp0` is the matlab I think.

**Pull request merged**